packages/node/src/core/providers/initialize.test.ts

```diff
@@ -47,13 +43,19 @@ import * as providers from './initialize';
 
 describe('initializeProviders', () => {
   it('sets the initial state for each provider', async () => {
-    const provider = new ethers.providers.JsonRpcProvider();
-
-    const getBlockNumber = provider.getBlockNumber as jest.Mock;
-    getBlockNumber.mockResolvedValueOnce(123456);
-    getBlockNumber.mockResolvedValueOnce(987654);
+    const contract = new ethers.Contract('address', ['ABI']);
+    contract.getProviderAndBlockNumber.mockResolvedValueOnce({
+      admin: '0xadmin1',
+      blockNumber: ethers.BigNumber.from(123456),
+      xpub: '0xxpub1',
+    });
+    contract.getProviderAndBlockNumber.mockResolvedValueOnce({
+      admin: '0xadmin2',
+      blockNumber: ethers.BigNumber.from(987654),
+      xpub: '0xxpub2',
+    });
 
-    const getLogs = provider.getLogs as jest.Mock;
+    const getLogs = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getLogs');
```

Comment:
It's better to use `spyOn` rather than mocking the entire module, then unmocking everything again with `jest.requireActual()` except for certain modules. Not necessary to update the older code, it's just something I've noticed.

Reply:
I haven't tried to fix this for the `ethers.Contract` parts yet though. Just had problems with this line so that's why I fixed this specifically